### PR TITLE
Liquidity baking doc and permit integration tests

### PR DESCRIPTION
**Pull request by claudebarde**
This pull request includes the new documentation for the liquidity baking contract and the update of the code for the permit integration tests.

### CODE REVIEW

File: integration-tests/contract-permits.spec.ts [hunk 1]
```diff
@@ -13,6 +14,73 @@ const errors_to_missigned_bytes = (errors: any[]) => {
...
+          { prim: "address" }
+        ]
+      },
+      {
+        prim: "pair",
+        args: [{ prim: "nat" }, { prim: "bytes" }]
+      }
+    ]
+  };
+  // const sigParamPacked = await Tezos.rpc.packData({
```
Comment (michaelkernaghan) [inline integration-tests/contract-permits.spec.ts:73]: can the commented-out code that is not an instruction to the user be removed?

File: integration-tests/contract-permits.spec.ts [hunk 2]
```diff
@@ -52,16 +120,17 @@ CONFIGS().forEach(({ lib, rpc, setup, createAddress }) => {
       });
       const packed_param = raw_packed.packed;
       const param_hash = buf2hex(blake.blake2b(hex2buf(packed_param), null, 32));
-      const bytes_to_sign = await permit_contract.methods
+      /*const bytes_to_sign = await permit_contract.methods
```
Comment (michaelkernaghan) [inline integration-tests/contract-permits.spec.ts:123]: is this commented out?

Comment (claudebarde) [reply integration-tests/contract-permits.spec.ts]: Yes, I updated the PR and removed the commented-out code

File: docs/liquidity_baking.md [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+- __%tokenToXtz__:
+```ts
+import { TezosToolkit } from "@taquito/taquito"
+
+// to take into account the subsidy added to the LB contract
+// when the transaction will be baked
+const creditSubsidy = (xtzPool: BigNumber | number): BigNumber => {
+    if (BigNumber.isBigNumber(xtzPool)) {
+      return xtzPool.plus(new BigNumber(2500000));
```
Comment (roxaneletourneau) [inline docs/liquidity_baking.md:68]: To clarify what the 2500000 refers to, consider defining a constant (ie: liquidityBakingSubsidy = 2500000)

File: docs/liquidity_baking.md [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+      tokenPool_.isGreaterThan(0)
+    ) {
+      // Includes 0.1% fee and 0.1% burn calculated separatedly: 
+      // 999/1000 * 999/1000 = 998100/1000000
+      let numerator = new BigNumber(tokenIn)
+        .times(new BigNumber(xtzPool))
+        .times(new BigNumber(998001));
+      let denominator = new BigNumber(tokenPool)
+        .times(new BigNumber(1000000))
+        .plus(new BigNumber(tokenIn).times(new BigNumber(999000)));
```
[the remark this answers is not in the document]
Comment (claudebarde) [reply docs/liquidity_baking.md:103]: This function was adapted from this code: https://gitlab.com/sophiagold/dexter-calculations/-/blob/liquidity_baking/index-raw.js#L390, I have no idea what 999000 is but I trust the great minds that wrote this code 😅

[the remark this answers is not in the document]
Comment (roxaneletourneau) [reply docs/liquidity_baking.md:104]: > the formula is (1 - burnFee) * (tezPool * tokenAmount * (1 - dexFee) / (tokenPool + tokenAmount * (1 - dexFee)))

File: docs/liquidity_baking.md [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+const lbContract = await Tezos.wallet.at(LB_CONTRACT_ADDRESS);
+const tzBtcContract = await Tezos.wallet.at(TZBTC_ADDRESS);
+const maxTokensSold = Math.floor(
+    AMOUNT_IN_TZBTC + (AMOUNT_IN_TZBTC * slippage) / 100
+);
+const minLqtMinted = Math.floor((AMOUNT_IN_XTZ * lqtTotal) / xtzPool);
+// the deadline value is arbitrary and can be changed
+const deadline = new Date(Date.now() + 60000).toISOString();
+
+const batchOp = await $store.Tezos.wallet
```
Comment (roxaneletourneau) [inline docs/liquidity_baking.md:218]: `$store.` should be removed

**Review by roxaneletourneau — CHANGES_REQUESTED** [review]

File: docs/liquidity_baking.md (hunk 5 above)
Comment (claudebarde) [reply docs/liquidity_baking.md]: Removed!

**Pull request merged**